aws/resource_aws_appconfig_application.go

```diff
@@ -0,0 +1,177 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/appconfig"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsAppconfigApplication() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppconfigApplicationCreate,
+		Read:   resourceAwsAppconfigApplicationRead,
+		Update: resourceAwsAppconfigApplicationUpdate,
+		Delete: resourceAwsAppconfigApplicationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
```

Comment:
how did you figure out that these were the bounds of a valid name?

Reply:
I used the AppConfig API [reference guide](https://docs.aws.amazon.com/appconfig/2019-10-09/APIReference/appconfig-api.pdf) from AWS. For each endpoint they lay out all the request parameters and constraints.